Foundation/NSObject.swift

```diff
@@ -118,8 +118,44 @@ public class NSObject : NSObjectProtocol {
     internal var _cfTypeID: CFTypeID {
         return 0
     }
-}
+    
+    // TODO move these back into extensions once extension methods can be overriden
+    
+    // TODO: Could perhaps be an extension of NSCoding instead. The reason it is an extension of NSObject is the lack of default implementations on protocols in Objective-C.
```

Comment:
Since we don't have optional implementations of delegate methods we made all of the delegation methods have default implementations; so extending NSCoding with a default implementation is not unreasonable and really wouldn't incur a full API change (provided it really does make sense in the implementation)